share/ci/templates/build_sonar.yml

```diff
@@ -0,0 +1,32 @@
+# azure-pipelines template file
+# https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops
+
+# Note: This is currently Linux only
+
+parameters:
+  cxxCompiler: ''
+  cCompiler: ''
+
+steps:
+- bash: |
+    if [ "$CXXCOMPILER" ]; then
+        export CXX="$CXXCOMPILER"
+    fi
+
+    if [ "$CCOMPILER" ]; then
+        export CC="$CCOMPILER"
+    fi
+
+    build-wrapper-linux-x86-64 --out-dir bw_output make clean all
+  workingDirectory: _build
+  displayName: Build OCIO with build-wrapper
+
+- template: test.yml
+
+- bash: share/ci/scripts/run_gcov.sh
+  displayName: Generate code coverage report
+
+- bash: sonar-scanner -X -Dsonar.login=$SONAR_TOKEN
```

Comment:
Is there a way to break on new `blocker`?

Reply:
I think the recommended approach from Sonar is to rely on notifications (which you can subscribe to) and status badges (which I should add to our README) to indicate a failed quality gate. In our case, since we will do analysis once weekly, even a failed CI build in this case could go unnoticed (without CI failure notifications). If we were using analysis for PRs, I think it would be possible to block a PR merge.

Reply:
ok. There is no short-term need to slowdown the pull request for this improvement.